Animation using padding-block doesnt override set padding style

graouts opened the pull request:
#### f438e854500a424cbcef992bcfec923d0b7d26b8
<pre>
Animation using padding-block doesnt override set padding style
<a href="https://bugs.webkit.org/show_bug.cgi?id=258813">https://bugs.webkit.org/show_bug.cgi?id=258813</a>
rdar://112023856

Reviewed by Tim Nguyen.

We did not support animating the padding-block shorthand. To do so we must do two things:

1. in CSSPropertyAnimation.cpp we must actually mark that shorthand as animatable.
2. in transitionMatchesProperty() we must support the possibility that a property
   is a shorthand for logical properties.

I suppose we could make transitionMatchesProperty() re-entrant in some way such that shorthands
of shorthands could also work, but this will do for now.

We also add animation support for padding-inline since it&apos;s a closely-related shorthand.

* LayoutTests/imported/w3c/web-platform-tests/css/css-logical/animations/padding-block-interpolation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-logical/animations/padding-block-interpolation.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-logical/animations/padding-inline-interpolation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-logical/animations/padding-inline-interpolation.html: Added.
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/style/Styleable.cpp:
(WebCore::transitionMatchesProperty):

Canonical link: <a href="https://commits.webkit.org/265956@main">https://commits.webkit.org/265956@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d83ab18aa38e9385ec8a25092ae41a97e9ccf3ba

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12391 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12727 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/13037 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/14134 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11919 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/12451 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/15152 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/12745 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14589 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12555 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13311 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10481 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/14556 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10594 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/11226 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/18323 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11677 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11391 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14571 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11884 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9808 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/11113 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/11097 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15443 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1382 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11739 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/animation/CSSPropertyAnimation.cpp:
```diff
@@ -3937,7 +3938,6 @@ CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap()
         case CSSPropertyOverscrollBehaviorX:
         case CSSPropertyOverscrollBehaviorY:
-        case CSSPropertyPaddingBlock: // logical shorthand
         case CSSPropertyPaddingInline: // logical shorthand
```
Seems wrong that this whole change is specific to `padding-block` and applies to no other logical shorthand (`padding-inline`, `margin-block/inline`, etc.)

graouts replied:
I disagree. We got a bug report specifically about this property and this patch addresses this. Yes, a part of the source change will be beneficial to additional properties of that ilk, but a smaller change is my preference because we’ve gotten regressions in the last doing too much at once. We can add additional properties bit by bit.

nt1m replied:
I think it would make sense to at least do `padding-inline` and `padding-block` at the same time.

Because inconsistency in behavior could be considered as a bug.

graouts replied:
OK, that makes sense. I've updated the patch to also support `padding-inline`. I'll do the `margin` and `border` properties as followups.

cdumez committed:
Allow suspension with certain locked database files
https://bugs.webkit.org/show_bug.cgi?id=259078
rdar://97212284

Reviewed by Brent Fulgham.

I checked several apps:
1. MobileSafari
2. SafariViewController in App A
3. SafariViewController in App B
4. WKWebView in App C

I printed the database paths in these 4 apps for:
1. IndexedDB
2. LocalStorage
3. ITP

In all cases, the database paths were distinct (not shared between apps).
As a result, it should be safe to suspend while any of these 3 database
files are locked. No other process should access it while we're suspended.

For these 3 database files, I now use the `com.apple.runningboard.can-suspend-locked`
file attribute to allow suspension while locked instead of killing the
process.

* Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.cpp:
(WebCore::IDBServer::SQLiteIDBBackingStore::getOrEstablishDatabaseInfo):
* Source/WebCore/platform/sql/SQLiteDatabase.cpp:
(WebCore::SQLiteDatabase::open):
* Source/WebCore/platform/sql/SQLiteDatabase.h:
(WebCore::SQLiteDatabase::open):
* Source/WebKit/NetworkProcess/DatabaseUtilities.cpp:
(WebKit::DatabaseUtilities::openDatabaseAndCreateSchemaIfNecessary):
* Source/WebKit/NetworkProcess/storage/SQLiteStorageArea.cpp:
(WebKit::SQLiteStorageArea::prepareDatabase):

Canonical link: https://commits.webkit.org/265951@main

youennf committed:
Resync WPT WebCodec tests up to 0679447
https://bugs.webkit.org/show_bug.cgi?id=259058
rdar://112018159

Reviewed by Tim Nguyen and Eric Carlson.

Skipping HEVC tests until we have proper support.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/README.md:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/encoded-video-chunk.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/encoded-video-chunk.any.js:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/encoded-video-chunk.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/full-cycle-test.https.any.html:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/full-cycle-test.https.any.js:
(promise_setup.async const):
(promise_setup):
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/full-cycle-test.https.any.worker.html:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/full-cycle-test.https.any.worker_h265_annexb-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/full-cycle-test.https.any.worker_h265_hevc-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/full-cycle-test.https.any_h265_annexb-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/full-cycle-test.https.any_h265_hevc-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/h265.annexb: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/h265.mp4: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoDecoder-codec-specific.https.any.html:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoDecoder-codec-specific.https.any.js:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoDecoder-codec-specific.https.any.worker.html:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoDecoder-codec-specific.https.any.worker_h265_annexb-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoDecoder-codec-specific.https.any.worker_h265_hevc-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoDecoder-codec-specific.https.any_h265_annexb-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoDecoder-codec-specific.https.any_h265_hevc-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-construction.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-construction.any.js:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-construction.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/w3c-import.log:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/webcodecs/videoFrame-construction.any-expected.txt:

Canonical link: https://commits.webkit.org/265952@main

alanbaradlay committed:
(REGRESSION)[IFC] Content in nested inline boxes with decoration clone may overflow unexpectedly
https://bugs.webkit.org/show_bug.cgi?id=259110

Reviewed by Antti Koivisto.

This code incorrectly assumed 0 as empty hash map value for float type (while it is inf).

It causes the available space grow infinite when not all (nested) inline-boxes have
the decoration-break property set (e.g. <span class=has_decoration_break><span>some content here</span></span>)
as the inner inline-box is not in the hashmap (and inlineBoxListWithClonedDecorationEnd.get() returns inf).

* LayoutTests/fast/inline/nested-inline-boxes-with-decoration-clone-expected.html: Added.
* LayoutTests/fast/inline/nested-inline-boxes-with-decoration-clone.html: Added.
* Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp:
(WebCore::Layout::availableWidth):

Canonical link: https://commits.webkit.org/265953@main

JonWBedard committed:
[buildbot] Enable autoinstaller when uploading files
https://bugs.webkit.org/show_bug.cgi?id=259114
rdar://112087736

Reviewed by Aakash Jain.

* Tools/Scripts/upload-file-to-url: Import webkitpy to enable autoinstaller.

Canonical link: https://commits.webkit.org/265954@main

TingPing committed:
[GLIB] Update env var used to disable using flatpak to build
https://bugs.webkit.org/show_bug.cgi?id=259116

Reviewed by Nikolas Zimmermann.

This env var is more clear in its purpose. WKDEV_SDK was used
by another script but was never an ideal choice.

* Tools/Scripts/webkitdirs.pm:
(shouldUseFlatpak):

Canonical link: https://commits.webkit.org/265955@main

graouts committed:
Animation using padding-block doesnt override set padding style
https://bugs.webkit.org/show_bug.cgi?id=258813
rdar://112023856

Reviewed by Tim Nguyen.

We did not support animating the padding-block shorthand. To do so we must do two things:

1. in CSSPropertyAnimation.cpp we must actually mark that shorthand as animatable.
2. in transitionMatchesProperty() we must support the possibility that a property
   is a shorthand for logical properties.

I suppose we could make transitionMatchesProperty() re-entrant in some way such that shorthands
of shorthands could also work, but this will do for now.

We also add animation support for padding-inline since it's a closely-related shorthand.

* LayoutTests/imported/w3c/web-platform-tests/css/css-logical/animations/padding-block-interpolation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-logical/animations/padding-block-interpolation.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-logical/animations/padding-inline-interpolation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-logical/animations/padding-inline-interpolation.html: Added.
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/style/Styleable.cpp:
(WebCore::transitionMatchesProperty):

Canonical link: https://commits.webkit.org/265956@main
